test/ng/directive/selectSpec.js

```diff
@@ -1123,5 +1123,19 @@ describe('select', function() {
       }).toThrowMinErr('ng','badname', 'hasOwnProperty is not a valid "option value" name');
     });
 
+    it('should set providedEmptyOption.selected to false', function() {
```

Comment:
> ngOptions will set an incorrect value to selected.value

Yes, `select.value` being wrong is end result.  More precisely, the value selected by `ng-model` is not reflected in the UI.  But it's not `ng-options` setting it incorrectly, the problem is caused by Firefox apparently ignoring the assignment `selected.value = $modelVaule` under some conditions.

I do not know precisely what those conditions are, but it appears this sequence of events is necessary but not sufficient:

 * We start out with select having one option, which is of course selected.
 * That option (which happens to be angular's EmptyOption) is then disabled (via `ng-disabled` in my case).
 *  `ng-options` then populates the options.  Firefox automagically selects the first one added because the only existing option is disabled.
 * The select controller then assigns `select.value`, but if you print it's value in a debuuger it appears this assignment has been ignored.  If you click the select box in the UI, both the 1st option and the new selected option are highlighted despite the select not being a multiselect.

When I added the code to remove both the `selected` attribute and `selected` property the bug in Firefox is no longer triggered.